graphql_compiler/tests/test_macro_schema.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+import unittest
+
+from graphql.utils.schema_printer import print_schema
+
+from ..macros import get_schema_with_macros
+from .test_helpers import get_empty_test_macro_registry, get_test_macro_registry
+
+
+class MacroSchemaTests(unittest.TestCase):
+    def setUp(self):
+        """Disable max diff limits for all tests."""
+        self.maxDiff = None
+        self.macro_registry = get_test_macro_registry()
+
+    def test_get_schema_with_macros_original_schema_unchanged(self):
+        empty_macro_registry = get_empty_test_macro_registry()
+        original_printed_schema = print_schema(self.macro_registry.schema_without_macros)
+        printed_schema_with_0_macros = print_schema(get_schema_with_macros(empty_macro_registry))
+        printed_schema_afterwards = print_schema(self.macro_registry.schema_without_macros)
+        self.assertEqual(original_printed_schema, printed_schema_afterwards)
+        self.assertEqual(original_printed_schema, printed_schema_with_0_macros)
+
+    def test_get_schema_with_macros_basic(self):
```

Comment:
Can you a slightly more complex test? For this test the object, parent and grandparent are all of the same type so the test will pass if the type is of any of the three.

Reply:
@pmantica1 thoughts on addressing this in a subsequent PR, so we can merge this, upload a new pre-release, and unblock work in other projects while we improve the test suite here?